279 - QA: FIlter Rename Rna-seq to RNA-seq

nozomione opened the pull request:
## Issue Number

Closing #279

## Purpose/Implementation Notes

Rename the `Rna-seq`name  to be `RNA-seq` in UI.

@dvenprasad Please see the latest UI [here](https://refinebio-b1uzwah0s-ccdl.vercel.app/).

## Types of changes

What types of changes does your code introduce?

<!-- Remove any which your PR isn't -->

- Bugfix (non-breaking change which fixes an issue)

## Functional tests

List out the functional tests you've completed to verify your changes work locally.

## Checklist

<!-- Put an `x` in the boxes that apply. -->

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

dvenprasad reviewed (approved):
👍 

🐮 🐮 🐮

davidsmejia reviewed (commented):
I'd like to see some changes before we merge this:

- move the `getFormattedTechnologyNames` to `helpers/formatTechnologyName`
- remove the nested ternary operator here and handle this by either:
  - test for default case first ie. `!['platform', 'technology'].includes(filterOption) and then short circuit for the specific special cases
  - alternatively, move this logic to a function at the top of the component that takes `filterOption` and `option`

Additionally, it looks like `helpers/formatPlatformName` correctly falls back to the passed in value so we can remove the `|| option[0]` fall back.

Looking at `helpers/formatPlatformName`, you could further clean it up by setting the default value to an empty string and removing that first check but that change doesnt need to occur in this PR.

nozomione commented:
> I'd like to see some changes before we merge this:
> 
>     * move the `getFormattedTechnologyNames` to `helpers/formatTechnologyName`
> 
>     * remove the nested ternary operator here and handle this by either:
>       
>       * test for default case first ie. `!['platform', 'technology'].includes(filterOption) and then short circuit for the specific special cases
>       * alternatively, move this logic to a function at the top of the component that takes `filterOption` and `option`
> 
> 
> Additionally, it looks like `helpers/formatPlatformName` correctly falls back to the passed in value so we can remove the `|| option[0]` fall back.
> 
> Looking at `helpers/formatPlatformName`, you could further clean it up by setting the default value to an empty string and removing that first check but that change doesnt need to occur in this PR.

Thank you for the idea💡! 

I made the following adjustments:
- moved the `getFormattedTechnologyNames` to `helpers/formatTechnologyName`
- added the new `helpers/formatFilterNames` which is responsible for formatting all the filter names for UI
   - defined the constants in `config/options` for the filter option names that need to be formatted 
- used the`helpers/formatFilterNames` in the `SearchFilter` component and cleaned up the ternary operators for filter names

Ready for another look, thank you!

davidsmejia commented on src/helpers/formatFilterNames.js:
```diff
@@ -0,0 +1,16 @@
+import { options } from 'config'
```
```suggestion
```

davidsmejia commented on src/components/SearchResults/SearchFilterList/SearchFilter.js:
```diff
@@ -95,10 +95,11 @@ export const SearchFilter = ({
-                        ? formatPlatformName(cache.platforms[option[0]]) ||
-                          option[0]
-                        : formatString(option[0])}
+                      {formatFilterNames(
+                        filterOption,
+                        option[0],
+                        cache.platforms[option[0]]
+                      )}
```
```suggestion
                      {formatFilterName(filterOption, option[0])}
```

davidsmejia commented on src/components/SearchResults/SearchFilterList/SearchFilter.js:
```diff
@@ -4,9 +4,8 @@ import styled, { css } from 'styled-components'
 import { useResponsive } from 'hooks/useResponsive'
 import { useSearchManager } from 'hooks/useSearchManager'
 import { TextHighlightContextProvider } from 'contexts/TextHighlightContext'
+import formatFilterNames from 'helpers/formatFilterNames'
```
```suggestion
import formatFilterName from 'helpers/formatFilterName'
```

davidsmejia commented on src/components/SearchResults/SearchFilterList/SearchFilter.js:
```diff
@@ -47,9 +46,10 @@ export const SearchFilter = ({
-            (filterOption === 'platform'
-              ? formatPlatformName(cache.platforms[option[0]]) || option[0]
-              : formatString(option[0])
+            formatFilterNames(
+              filterOption,
+              option[0],
+              cache.platforms[option[0]]
             )
```
```suggestion
            formatFilterName(
              filterOption,
              option[0]
            )
```

davidsmejia commented on src/helpers/formatTechnologyName.js:
```diff
@@ -0,0 +1,7 @@
+import formatString from 'helpers/formatString'
+
+export default (option) => {
+  const rnaSeq = 'RNA-seq' // 'rna-seq' should be rendered as 'RNA-seq' in UI
+
+  return option.match(rnaSeq.toLowerCase()) ? rnaSeq : formatString(option)
+}
```
`technologyNames` would make sense to put in config now

```suggestion
const technologyNames = {
  'rna-seq': 'RNA-seq'
}
export default (option) => {
  return technologyNames[option.toLowerCase()] || formatString(option)
}
```

davidsmejia commented on src/helpers/formatFilterNames.js:
```diff
@@ -0,0 +1,16 @@
+import formatPlatformName from 'helpers/formatPlatformName'
+import formatTechnologyName from 'helpers/formatTechnologyName'
+
+export default (filterOption, item, formattedPlatformName) => {
+  const { filterOptionsToFormat } = options.search
+  const formatName = () =>
+    filterOption === 'platforom'
+      ? formatPlatformName(formattedPlatformName)
+      : formatTechnologyName(item)
+
+  return !filterOptionsToFormat[filterOption]
+    ? formatString(item)
+    : formatName()
+}
```
I think we can make this easier to understand. This function really just needs to take a `filterOption` and map that to a corresponding formatting function. 

This means that we will need to move the cache.platforms logic to the `formatPlatformName` helper. That function should still just take `platformName` as the only argument only now it will check to see if it is in the cache then handle the value the same as it currently does.

```suggestion
export default (filterOption, name) => {
  
  if (filterOption === 'platform') {
    return formatPlatformName(name)
  }
  
  if (filterOption === 'technology') {
    return formatTechnologyName(name)  
  }

  return formatString(item)
}
```

nozomione replied:
Now using the API cache directly in `helpers/formatPlatformName` and refactored `helpers/formatFilterName` with the switch statement for readability and maintainability. Thank you for the feedback!

davidsmejia reviewed (commented):
Can we make the helper to `formatFilterName` since it only handles a single name.

davidsmejia reviewed (approved):
LGTM
